tools/eclipse_prj_helper/Makefile.template

```diff
@@ -0,0 +1,47 @@
+###########################################################################
+#
+#  Copyright (c) 2013-2017, ARM Limited, All Rights Reserved
+#  SPDX-License-Identifier: Apache-2.0
+#
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+###########################################################################
+
+###########################################################################
+#            DO NOT EDIT - This is an auto-generated file                 #
+###########################################################################
+# generated from:
+#   $template_file
+
+
+WORKSPACE_DIR := $workspace_dir
+UVISOR_ROOT_DIR := $uvisor_dir
+IMPORTER_DIR := $importer_dir
+
+PLATFORM := $target
+TOOLCHAIN := $toolchain
+
+
+
+.PHONY: all clean
+all:
+	@echo Build uVisor
+	@test -e $(UVISOR_ROOT_DIR) && $(MAKE) -C $(UVISOR_ROOT_DIR)
+	@echo Publish uVisor build artifacts
+	@test -e $(UVISOR_ROOT_DIR) && $(MAKE) -C $(IMPORTER_DIR) publish
+	@echo Build Workspace
+	@cd $(WORKSPACE_DIR) && mbed compile --profile debug -m $(PLATFORM) -t $(TOOLCHAIN)
```

Comment:
@Patater - the make file will fail - in case any of it is recipes will fail. Thus application wont be built in case uVisor build was not successful.
@theotherjimmy - calling make from make is requirement for rebuilding uVisor sources - which happens to have make driven build system.
@theotherjimmy - regarding the purposed makefile change. The purposed changes has following flaws comparing to original ones:
1. the purposed change tries to mimic mbed-cli dependencies - mbed-cli is fully capable to manage its own build dependencies, which i tend to believe happen to more complicated than a simple glob. 
2. in case mbed-cli does rebuild the binary on every build regardless to its dependencies - the fix should be applied to mbed-cli rather than usage scripts.
3. the purposed change introduces dependencies which will not be honored in parralel build use case (-j) as there is no guarantee that `uvisor-publish` will be executed after `uvisor-executable`